src/tjoperators.jl

```diff
@@ -0,0 +1,639 @@
+#= Operators that act on t-J-type models
+i.e. the local hilbert space consists of 
+=#
+module TJOperators
+
+using TensorKit
+
+export tj_space
+export c_num, u_num, d_num, c_num_hole
+export S_x, S_y, S_z, S_plus, S_min
+export u_plus_u_min, d_plus_d_min
+export u_min_u_plus, d_min_d_plus
+export u_min_d_min, d_min_u_min
+export c_plus_c_min, c_min_c_plus, c_singlet
+export S_plusmin, S_minplus, S_exchange
+
+export nꜛ, nꜜ, nʰ, n
+export Sˣ, Sʸ, Sᶻ, S⁺, S⁻
+export u⁺u⁻, d⁺d⁻, u⁻u⁺, d⁻d⁺, u⁻d⁻, d⁻u⁻
+export c⁺c⁻, c⁻c⁺
+export S⁻⁺, S⁺⁻
+
+"""
+    tj_space(particle_symmetry::Type{<:Sector}, spin_symmetry::Type{<:Sector}; slave_fermion::Bool = false)
+
+Return the local hilbert space for a t-J-type model with the given particle and spin symmetries.
+The possible symmetries are 
+- Particle number: `Trivial`, `U1Irrep`;
+- Spin: `Trivial`, `U1Irrep`, `SU2Irrep`.
+
+Setting `slave_fermion = true` switches to the slave-fermion basis. 
+
+- basis states for `slave_fermion = false`: 
+    |∅⟩, |↑⟩, |↓⟩
```

Comment:
I would like to slightly modify the notation of the basis states to be more consistent with docstrings below. 
```suggestion
    |0⟩ = |vac⟩ (vacuum), |↑⟩ = (c↑)†|vac⟩, |↓⟩ = (c↓)†|vac⟩
```

Reply:
I'm okay with either, but I did like the single character ket: `|0⟩` or `|∅⟩` both look slightly nicer to me. No strong preferences, and definitely agree that they should be consistent.